geotrek/tourism/templates/tourism/sql/post_20_views.sql

```diff
@@ -137,18 +137,18 @@ LEFT JOIN
      GROUP BY c.id) p ON a.id = p.id
 LEFT JOIN authent_structure c ON a.structure_id = c.id
 LEFT JOIN common_reservationsystem d ON a.reservation_system_id = d.id
-LEFT JOIN
-    (SELECT array_to_string(ARRAY_AGG (b.name ORDER BY b.name), ', ', '_') zoning_city,
-            a.id
-     FROM tourism_touristiccontent a
-     JOIN zoning_city b ON ST_INTERSECTS (a.geom, b.geom)
-     GROUP BY a.id) f ON a.id = f.id
-LEFT JOIN
-    (SELECT array_to_string(ARRAY_AGG (b.name ORDER BY b.name), ', ', '_') zoning_district,
-            a.id
-     FROM tourism_touristiccontent a
-     JOIN zoning_district b ON ST_INTERSECTS (a.geom, b.geom)
-     GROUP BY a.id) g ON a.id = g.id
+LEFT JOIN LATERAL (
+     SELECT array_to_string(array_agg(b_1.name ORDER BY b_1.name), ', '::text, '_'::text) AS zoning_city
+           FROM   zoning_city b_1
+            WHERE st_intersects(a.geom, b_1.geom)
+          GROUP BY a.id
+    ) f ON true
```

Comment:
**Suggestion:** Consider using `ST_DWithin` instead of `ST_Intersects` for spatial intersection checks. `ST_DWithin` can provide better performance as it doesn't need to calculate the actual intersection geometry. [performance]
```suggestion
LEFT JOIN LATERAL (
     SELECT array_to_string(array_agg(b_1.name ORDER BY b_1.name), ', '::text, '_'::text) AS zoning_city
           FROM   zoning_city b_1
            WHERE st_dwithin(a.geom, b_1.geom, 0)
          GROUP BY a.id
    ) f ON true
```